### PR TITLE
Workspace security fixing

### PR DESCRIPTION
**Pull request by pysailor**
- Take away the View permission from Authenticated in "private" workspaces.
  This currently violates "Workspace is visible to non-members", but since we have no story to request access, we leave it like this
- Do not show buttons / links for creating content to users who do not have the permission to use them
- In the social activity stream, make sure comments are shown to all with viewer permissions, but only the commenting form is hidden 
- Add permission "Add portal content" to the workflow of workspaces to turn off inheritance
  This permission is given, in all states of the workflow, to role Contributor
- Tighten security in microblog: If the context of a StatusUpdate is a workspace, and that workspace cannot be resolved because the user does not have the permission to view it, don't silently ignore but raise an error.

### CODE REVIEW

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/141/
Test FAILed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/142/
Test FAILed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/143/
Test PASSed.

File: src/ploneintranet/workspace/browser/workspace.py [hunk 1]
```diff
@@ -26,6 +26,13 @@ def can_manage_workspace(self):
             obj=self,
         )
 
+    def can_add_status_updates(self):
+        """
+        Does this user have the permission to add status updates
+        """
+        return api.user.has_permission(
+            "Plone Social: Add Microblog Status Update", obj=self)
```
Comment (gyst) [inline src/ploneintranet/workspace/browser/workspace.py]: Shouldn't this be self.context?

File: src/ploneintranet/workspace/browser/tiles/workspaces.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ def my_workspaces(context):
             })
         except Unauthorized:
```
Comment (gyst) [inline src/ploneintranet/workspace/browser/tiles/workspaces.py]: This whole try/except can now be dropped - that was one of the goals of this story, to be able to rely on the portal_catalog security filtering.

File: src/ploneintranet/workspace/profiles/default/workflows/ploneintranet_workspace_workflow/definition.xml [hunk 3]
```diff
@@ -31,6 +32,12 @@
       <permission-role>Site Administrator</permission-role>
       <permission-role>TeamManager</permission-role>
     </permission-map>
+    <permission-map name="Add portal content" acquired="False">
+      <permission-role>Manager</permission-role>
+      <permission-role>Site Administrator</permission-role>
+      <permission-role>TeamManager</permission-role>
+      <permission-role>Contributor</permission-role>
+    </permission-map>
```
Comment (gyst) [inline src/ploneintranet/workspace/profiles/default/workflows/ploneintranet_workspace_workflow/definition.xml:40]: We should add owner here and in several other places but that's scheduled as a separate story Assembla 430.

File: src/ploneintranet/workspace/tests/test_workspace_workflow.py [hunk 4]
```diff
@@ -67,8 +70,8 @@ def test_private_workspace(self):
             username='nonmember',
             obj=self.workspace_folder,
         )
-        self.assertTrue(nonmember_permissions[VIEW],
-                        'Non-member cannot view private workspace')
+        self.assertFalse(nonmember_permissions[VIEW],
+                         'Non-member cannot view private workspace')
```
Comment (gyst) [inline src/ploneintranet/workspace/tests/test_workspace_workflow.py]: The error message on test fail should become "can" not "cannot".

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/144/
Test PASSed.

**Comment by gyst** [comment]
@pysailor You missed one of my comments on src/ploneintranet/workspace/browser/workspace.py

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/147/
Test PASSed.

**Pull request merged**